src/BinLinker.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace MariusBuescher\NodeComposer;
+
+use Composer\Util\Filesystem;
+use Composer\Util\ProcessExecutor;
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem as SymfonyFilesystem;
+
+/**
+ * Class to symlink bin files
+ *
+ * @package MariusBuescher\NodeComposer
+ */
+class BinLinker
+{
+    /**
+     * Composer vendor bin dir
+     *
+     * @var string
+     */
+    protected $vendorBinDir;
+
+    /**
+     * Filesystem instance for operations with files
+     *
+     * @var SymfonyFilesystem
+     */
+    protected $filesystem;
+
+    /**
+     * Os type
+     *
+     * @var string
+     */
+    private $osType;
+
+    /**
+     * Filesystem constructor.
+     *
+     * @param string $vendorBinDir Where all bin files should be kept?
+     * @param string $osType OS type
+     */
+    public function __construct($vendorBinDir, $osType)
+    {
+        $this->vendorBinDir = $vendorBinDir;
+        $this->osType = strtolower($osType);
+        $this->filesystem = new SymfonyFilesystem();
+    }
+
+    /**
+     * Links bin item
+     *
+     * @param string $from Path from where to create symlink
+     * @param string $to Path to where create symlink
+     */
+    public function linkBin($from, $to)
+    {
+        if ($this->osType === 'win') {
+            $this->filesystem->dumpFile(
```

Comment:
This is not really a link. Isn't there another way of generating a link on windows?

Reply:
Sadly, on windows symlink creation for some reason works only when command is running with administrator rights. And that is not very common between servers.

Reply:
Mmmh, OK. I think the last version of Windows I really used was XP, so I have no idea what is going on on these machines.